lib/chef/chef_class.rb

```diff
@@ -190,6 +190,22 @@ def provider_handler_map
     def resource_handler_map
       @resource_handler_map ||= Chef::Platform::ResourceHandlerMap.instance
     end
+
+    #
+    # @overload log
+    # Get the current log object.
+    #
+    # @return An object that supports `deprecation(message)`
+    #
+    # @example
+    #     run_context.log.deprecation("Deprecated!")
+    #
+    # @api private
+    def log
```

Comment:
Yeah, I agree at the moment; that's why it's `@api private`. The intent is that it actually return an object that has the same interface as `Chef::Log` though--it's just `run_context`-specific and can be intercepted by a formatter. I can rename it `deprecator` for now until that's implemented.